ti/api/query.js

```diff
@@ -1,86 +1,87 @@
 }
 
-function query(query, db = 'pitcher') {
+function query (query, db = 'pitcher') {
     return new Promise((resolve, reject) => {
-        if (hasCached(query)){
-            return resolve(cache[query].result);
+        if (hasCached(query)) {
+            return resolve(cache[query].result)
         }
-
         fireEvent('dbFunction', {
-                db: 'pitcher',
-                pType: 'query',
-                query: query,
-            })
-            .then(function (e) {
-                let result = [];
-
-                if (e.error) {
-                    reject(new Error(e.error))
-                }
-
-                for (let i = 0; i < e.results.length; i++) {
-                    let res = e.results[i];
-                    let obj = {};
+            db: 'pitcher',
+            pType: 'query',
+            query: query,
+        }).then(function (e) {
+            let result = []
+            if (e.error) {
+                reject(new Error(e.error))
+            }
+            for (let i = 0; i < e.results.length; i++) {
+                let res = e.results[i]
+                let obj = {}
 
-                    for (let j = 0; j < e.columns.length; j++) {
-                        let column = e.columns[j];
+                for (let j = 0; j < e.columns.length; j++) {
+                    let column = e.columns[j]
 
-                        if (column === 'extraField') {
-                            let o = JSON.parse(res[j]);
+                    if (column === 'extraField') {
+                        let o = JSON.parse(res[j])
 
-                            for (let n in o) {
-                                if (o.hasOwnProperty(n)) {
-                                    obj[n] = o[n];
-                                }
+                        for (let n in o) {
+                            if (o.hasOwnProperty(n)) {
+                                obj[n] = o[n]
                             }
+                        }
+                    } else {
+                        if (typeof res[j] == 'undefined') {
+                            obj[column] = null
                         } else {
-                            if (typeof res[j] == 'undefined') {
-                                obj[column] = null
-                            } else {
-                                obj[column] = res[j]
-                            }
+                            obj[column] = res[j]
                         }
                     }
-
-                    result.push(obj)
                 }
-
-                if (cacheEnabled) {
-                    cacheQuery(query, result);
-                }
-
-                resolve(result)
-            })
-            .catch(reject)
+                result.push(obj)
+            }
+            if (cacheEnabled) {
+                cacheQuery(query, result)
+            }
+            resolve(result)
+        }).catch(reject)
     })
 }
 
+function contextQuery (query, context) {
```

Comment:
Can we make the contextQuery a new file? I think it's unecessary to import HandleBars, if I run a simple query

Reply:
done